src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs

```diff
@@ -0,0 +1,510 @@
+
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.DirectoryServices.Tests;
+using System.Linq;
+using Xunit;
+
+namespace System.DirectoryServices.AccountManagement.Tests
+{
+    public class AccountManagementTests
+    {
+        internal static bool IsLdapConfigurationExist => LdapConfiguration.Configuration != null;
+        internal static bool IsActiveDirectoryServer => IsLdapConfigurationExist && LdapConfiguration.Configuration.IsActiveDirectoryServer;
+
+        [ConditionalFact(nameof(IsActiveDirectoryServer))]
+        public void TestCurrentUse()
+        {
+            using (PrincipalContext context = DomainContext)
+            using (UserPrincipal p = FindUser(LdapConfiguration.Configuration.UserNameWithNoDomain, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(LdapConfiguration.Configuration.UserNameWithNoDomain, p.Name);
+
+                using (UserPrincipal cu = UserPrincipal.Current)
+                {
+                    try
+                    {
+                        // UserPrincipal.Current get the context from the current thread.
+                        PrincipalContext pc = cu.Context;
+                        Assert.NotEqual(pc.Name, p.Context.Name);
+                    }
+                    catch
+                    {
+                        // ignore the exception here as UserPrincipal.Current.Context can throw for
+                        // non joined domain machines
+                    }
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(IsActiveDirectoryServer))]
+        public void TestCurrentUserUsingSearchFilter()
+        {
+            using (PrincipalContext context = DomainContext)
+            using (UserPrincipal p = FindUserUsingFilter(LdapConfiguration.Configuration.UserNameWithNoDomain, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(LdapConfiguration.Configuration.UserNameWithNoDomain, p.Name);
+            }
+        }
+
+        [ConditionalFact(nameof(IsActiveDirectoryServer))]
+        public void TestGuestsGroup()
+        {
+            using (PrincipalContext context = DomainContext)
+            using (GroupPrincipal p = FindGroup("Guests", context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal("Guests", p.Name);
+                Assert.Equal("Guests", p.SamAccountName);
+            }
+        }
+
+        [ConditionalFact(nameof(IsActiveDirectoryServer))]
+        public void TestAddingUser()
+        {
+            DeleteUser("corefxtest1");
+            DeleteUser("corefxtest2");
+            DeleteUser("corefxtest3");
+
+            try
+            {
+                using (PrincipalContext context = DomainContext)
+                using (UserPrincipal p1 = CreateUser(context, "corefxtest1", "netcore#1pass", "corefx1", "test1", "CoreFx Test User 1"))
```

Comment:
you've got names and credentials copied and hardcoded in a couple places here.  These should be replaced with a single source